Code/Framework/AzToolsFramework/Tests/PythonLoaderTests.cpp

```diff
@@ -0,0 +1,127 @@
+        AZ::IO::SystemFile::CreateDir(tempVenvFullPath.String().c_str());
+        //AZ::IO::FileIOBase::GetInstance()->CreatePath(tempVenvFullPath.String().c_str());
+        AZ::IO::FixedMaxPath tempPyConfigFile = tempVenvRelativePath / "pyvenv.cfg";
+        AZStd::string testPython3rdPartyPath = "/home/user/python/";
+        AZStd::string testPyConfigFileContent = AZStd::string::format("home = %s\n"
+                                                                      "include-system-site-packages = false\n"
+                                                                      "version = 3.10.13",
+                                                                      testPython3rdPartyPath.c_str());
+        AZ::Test::CreateTestFile(m_tempDirectory, tempPyConfigFile, testPyConfigFileContent);
+
+
+        // Test the method
+        AZ::IO::FixedMaxPath engineRoot{ s_testEnginePath };
+        AZ::IO::FixedMaxPath test3rdPartyRoot = m_tempDirectory.GetDirectoryAsFixedMaxPath() / s_test3rdPartySubPath;
+
+        auto result = AzToolsFramework::EmbeddedPython::PythonLoader::GetPythonExecutablePath(test3rdPartyRoot, engineRoot);
+        AZ::IO::FixedMaxPath expectedPath{ testPython3rdPartyPath };
+
+        EXPECT_TRUE(result == expectedPath);
+    }
+
+
+    TEST_F(AzToolsFrameworkPythonLoaderFixture, TestReadPythonEggLinkPaths_Valid)
+    {
+        // Prepare the test folder and create dummy egg-link files
+        AZ::IO::FixedMaxPath testRelativeSiteLIbsPath = AZ::IO::FixedMaxPath(s_test3rdPartySubPath) / "venv" / s_testEnginePathHashId / O3DE_PYTHON_SITE_PACKAGE_SUBPATH;
+        AZ::IO::FixedMaxPath testFullSiteLIbsPath = m_tempDirectory.GetDirectoryAsFixedMaxPath() / testRelativeSiteLIbsPath;
+        AZ::IO::SystemFile::CreateDir(testFullSiteLIbsPath.String().c_str());
+        //AZ::IO::FileIOBase::GetInstance()->CreatePath(testFullSiteLIbsPath.String().c_str());
+
+        AZStd::vector<AZStd::string> expectedResults;
+        expectedResults.emplace_back(testFullSiteLIbsPath.LexicallyNormal().Native());
+
+        //static constexpr const char* testEggLinkPaths[] = { "/lib/path/one", "/lib/path/two", "/lib/path/three" };
+        //size_t testEggLinkPathCount = sizeof(testEggLinkPaths) / sizeof(const char*);
+
+        static constexpr auto testEggLinkPaths = AZStd::to_array<const char*>({ "/lib/path/one", "/lib/path/two", "/lib/path/three" });
+        int index = 0;
+        for (const char* testEggLinkPath : testEggLinkPaths)
+        {
+            ++index;
+            AZStd::string testEggFileName = AZStd::string::format("test-%d.egg-link", static_cast<int>(index));
+            const char* lineBreak = ((index % 2) == 0) ? "\n" : "\r\n";
+            AZStd::string testEggFileContent = AZStd::string::format("%s%s.", testEggLinkPath, lineBreak);
+            expectedResults.emplace_back(AZStd::string(testEggLinkPath));
+
+            AZ::IO::FixedMaxPath testEggILeNamePath = testRelativeSiteLIbsPath / testEggFileName;
```

Comment:
A couple of typos here.
I believe the variable name of `testEggILeNamePath` should be `testEggLinkNamePath`?

Also the `testRelativeSiteLIbsPath`variable  has a capitial 'I' in the word LIbs.
```suggestion
            AZ::IO::FixedMaxPath testEggLinkNamePath = testRelativeSiteLibsPath / testEggFileName;
```